[subviews] Cursor sprite collecting garbage

This is a problem I just discovered on Linux - I'm pretty sure it wasn't there some time ago. Anyway, it's specific to subviews, my Linux computer and SDL2.

After moving the cursor around a bit, it starts to look like this.

![cursor](https://user-images.githubusercontent.com/238558/69189514-c7f70300-0b1e-11ea-8755-5e2dbf68bb76.jpg)

I try to bisect it later.

Oi, I'm getting this error, even with -DDISABLE_WERROR=1 (all the SDL2 dev packages are installed):

```
home/nadia/Software/Games/gemrb-subviews2019_11_19/gemrb/plugins/SDLVideo/SDL20Video.cpp:33:13: error: ‘SDL_BLENDOPERATION_ADD’ was not declared in this scope
             SDL_BLENDOPERATION_ADD, SDL_BLENDFACTOR_ZERO,
             ^~~~~~~~~~~~~~~~~~~~~~
/home/nadia/Software/Games/gemrb-subviews2019_11_19/gemrb/plugins/SDLVideo/SDL20Video.cpp:34:13: error: ‘SDL_BLENDFACTOR_ONE_MINUS_SRC_ALPHA’ was not declared in this scope
             SDL_BLENDFACTOR_ONE_MINUS_SRC_ALPHA, SDL_BLENDOPERATION_ADD);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/nadia/Software/Games/gemrb-subviews2019_11_19/gemrb/plugins/SDLVideo/SDL20Video.cpp:34:72: error: ‘SDL_ComposeCustomBlendMode’ was not declared in this scope
             SDL_BLENDFACTOR_ONE_MINUS_SRC_ALPHA, SDL_BLENDOPERATION_ADD
```

--- a/gemrb/plugins/SDLVideo/SDL20Video.cpp
+++ b/gemrb/plugins/SDLVideo/SDL20Video.cpp
@@ -128,7 +128,7 @@ int SDL20VideoDriver::UpdateRenderTarget(const Color* color)
 
 	if (screenClip.Dimensions() == screenSize)
 	{
-		// SDL does _not_ seem to like having a clip rect of the entire renderer size
+		// Some SDL backends complain on having a clip rect of the entire renderer size
 		// I'm not sure if it is an SDL bug; possibly its just 0 based so it is out of bounds?
 		SDL_RenderSetClipRect(renderer, NULL);
 	} else {

--- a/gemrb/plugins/SDLVideo/SDL20Video.h
+++ b/gemrb/plugins/SDLVideo/SDL20Video.h
@@ -163,6 +163,14 @@ public:
 	void Clear() {
 		SDL_SetRenderTarget(renderer, texture);
 		SDL_SetRenderDrawColor(renderer, 0, 0, 0, SDL_ALPHA_TRANSPARENT);
+#if SDL_COMPILEDVERSION == SDL_VERSIONNUM(2, 0, 10)
+		/**
+		 * See GH issue #410. In some SDL2 backends of this version, a clear
+		 * runs over an outdated state of the clipping settings. This can
+		 * be overcome by a harmless draw command right before.
+		 */
+		SDL_RenderDrawPoint(renderer, 0, 0);
+#endif
 		SDL_RenderClear(renderer);
 
 		ClearMaskLayer();
@@ -172,6 +180,9 @@ public:
 		if (maskLayer) {
 			SDL_SetRenderTarget(renderer, maskLayer);
 			SDL_SetRenderDrawColor(renderer, 0, 0, 0, SDL_ALPHA_TRANSPARENT);
+#if SDL_COMPILEDVERSION == SDL_VERSIONNUM(2, 0, 10)
+			SDL_RenderDrawPoint(renderer, 0, 0);
+#endif
 			SDL_RenderClear(renderer);
 		}
 	}

